src/event/event_epoll.c

```diff
@@ -89,12 +89,12 @@ _dispatch_muxnote_bucket(int ident)
 	return &_dispatch_sources[DSL_HASH((uint32_t)ident)];
 }
 #define _dispatch_unote_muxnote_bucket(du) \
-	_dispatch_muxnote_bucket(du._du->du_ident)
+	_dispatch_muxnote_bucket((int)du._du->du_ident)
 
 DISPATCH_ALWAYS_INLINE
 static inline dispatch_muxnote_t
 _dispatch_muxnote_find(struct dispatch_muxnote_bucket_s *dmb,
-		uint64_t ident, int16_t filter)
+		int ident, int16_t filter)
```

Comment:
my worry is that signed stuff for a hash table is a bad idea. Moving to `uint32_t` would be fine fWIW. and the reason for `uint32_t` for `du_ident`is because it's unsigned on Darwin because of kevent and it'll be hell if we change signedness of this field.